app/components/machine/driver-aliyunecs/component.js

```diff
@@ -10,23 +10,585 @@ let ioOptimized=[
+    description: "ecs.c4.2xlarge - 8 CPU,16 GB memory,3 Gbps network bandwidth,400k PPS",
+  },
+  {
+    value: "ecs.c4.4xlarge",
+    description: "ecs.c4.4xlarge - 16 CPU,32 GB memory,6 Gbps network bandwidth,800k PPS",
+  },
+];
+
 export default Ember.Component.extend(Driver, {
   driverName       : 'aliyunecs',
   aliyunecsConfig      : Ember.computed.alias('model.aliyunecsConfig'),
   ioOptimized : ioOptimized,
+  instanceType: instanceType,
+  dataDiskCategory: dataDiskCategory,
+  systemDiskCategory: systemDiskCategory,
+  regions: regions,
 
-
+  ioOptimizedObserves: function(){
+    if (this.get('aliyunecsConfig.ioOptimized')==='none'){
+      this.set('instanceType',notOptimizedinstanceType);
+      this.set('dataDiskCategory',[{value:"cloud"}]);
+      this.set('systemDiskCategory',[{value:"cloud"}]);
+    }
+    else {
+      this.set('instanceType',isOptimizedinstanceType);
+      this.set('dataDiskCategory',optimizedDiskCategory);
+      this.set('systemDiskCategory',optimizedDiskCategory);
+    }
+  }.observes('aliyunecsConfig.ioOptimized'),
   bootstrap: function() {
     let config = this.get('store').createRecord({
       type                  : 'aliyunecsConfig',
     });
-
+    this.set('instanceType',notOptimizedinstanceType);
     this.set('model', this.get('store').createRecord({
       type: 'host',
       aliyunecsConfig: config,
     }));
-
+    this.set('model.engineRegistryMirror',['https://s06nkgus.mirror.aliyuncs.com']);
+    this.set('model.engineInstallUrl','http://dev-tool.oss-cn-shenzhen.aliyuncs.com/docker-install/1.12.6.sh');
     //this.set('editing', false);
   },
+  validate: function(){
+    this._super();
+    var errors = this.get('errors') || [];
+    var name = this.get('model.hostname');
+    if (name) {
+      //name length rules
+      if (name.length<2){
+        errors.push("Name should have more then 2 letters.");
+      }
+      //check '.','-' can not be used as the first letter and the last letter
+      if (!/^[a-zA-Z0-9]([\.-]?[a-zA-Z0-9]+[\.-]?)*[a-zA-Z0-9]$/.test(name)){
+        errors.push("Name doesn't match the rule of Aliyunecs hostname rule");
+      }
+      //windows hostname can not have more then 15 letter
 
+      //linux hostname can not have more then 30 letter
+      if (name.length>30){
+        errors.push("The hostname can not have more then 30 letters.");
+      }
+    }
+    var accessKey = this.get('aliyunecsConfig.accessKeyId');
+    var accessSecret = this.get('aliyunecsConfig.accessKeySecret');
+    if (!accessKey){
+      errors.push("Access key is required.");
+    }
+    if (!accessSecret){
+      errors.push("Access Secret is required");
+    }
+    var sshPassword = this.get('aliyunecsConfig.sshPassword');
+    if (!sshPassword){
+      errors.push("SSH Password is required");
+    }
+    if (sshPassword && (sshPassword.length<8) || sshPassword.length>30){
+      errors.push("SSH Password should have at least 8 chatacter and less then 30 chatacter");
```

Comment:
"should be between 8 and 30 characters".

(Also for other messsages: "more than", not "more then")